kostyor/upgrades/engine.py

```diff
@@ -152,17 +152,21 @@ def start(self):
         # keystone, then with nova, and so on. See iteration details in
         # get_controllers() docstring.
         for host in get_controllers(hosts):
+            self.driver.pre_host_upgrade_hook(host)
```

Comment:
I mean having

```python
subtasks.append(self.driver.pre_host_upgrade_hook(host))
```

just in case `pre_host_upgrade_hook` wants to execute some long-running task. I think if driver doesn't need any pre-host upgrade hook, it will return some `NoopTask`.  Otherwise, we are limited to have quick tasks in order to do not block the process.